homeassistant/components/heos/media_player.py

```diff
@@ -0,0 +1,160 @@
+"""Denon HEOS Media Player."""
+
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    DOMAIN, MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA, SUPPORT_PREVIOUS_TRACK, SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP)
+from homeassistant.const import STATE_OFF, STATE_PAUSED, STATE_PLAYING
+
+from . import DOMAIN as HEOS_DOMAIN
+
+DEPENDENCIES = ['heos']
+
+SUPPORT_HEOS = SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_PAUSE | \
+        SUPPORT_PLAY_MEDIA | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+        SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP
+
+PLAY_STATE_TO_STATE = {
+    'play': STATE_PLAYING,
+    'pause': STATE_PAUSED,
+    'stop': STATE_OFF
```

Comment:
Is this a correct way to turn off the media player? There is an issue in the current custom component that doesn't allow you to turn HEOS back on once it's stopped. Zhe made a solution to basically never turn HEOS off, only pause it (more on this here: https://community.home-assistant.io/t/denos-heos-support/2465/71)  
```suggestion
    'stop': STATE_PAUSED
```

Reply:
No, that isn't the right change.  HEOS products  (non-receivers) only have 3 states (`play`, `pause`, and `stop`).  This should be mapped to `STATE_IDLE` instead.  Like Sonos, they have a physical power switch, but that completely turns off the device and disconnects it from the network.  These devices are typically always on.  Receivers have on/off through the API, but that's actually toggling stand-by mode (the network module is still on and connected).

As far as I can tell, there's no HEOS CLI control for power state -- like the zone information, I'm guessing that is done via the API to the receiver directly.  When a receiver is off, does it not turn on when you try to "Play" through the integration?  I haven't tested that myself, but I can explore that this evening.

Reply:
Yep, you are right, it will auto turn on receivers / speakers if they are off, didn't cross my mind to do state_idle. Nicely done!